File: androidtoolkit/src/main/java/net/oschina/git/roland/androidtoolkit/tasksubmit/BaseTaskStep.java
package net.oschina.git.roland.androidtoolkit.tasksubmit;

/**
 * Created by Roland on 2017/5/17.
 */

public abstract class BaseTaskStep {

    protected StepProgressListener progressListener;

    protected boolean result = false;

    public abstract void start();

    protected void finish(boolean result) {
        this.result = result;
        if (progressListener != null) {
            progressListener.onStepFinish(this);
        }
    }

    public void setProgressListener(StepProgressListener listener) {
        this.progressListener = listener;
    }

    public boolean getResult() {
        return result;
    }
}

File: androidtoolkit/src/main/java/net/oschina/git/roland/androidtoolkit/tasksubmit/BaseTaskStep.kt
package net.oschina.git.roland.androidtoolkit.tasksubmit

/**
 * Created by Roland on 2017/5/17.
 */

abstract class BaseTaskStep {

    var progressListener: StepProgressListener? = null

    var result = false
        protected set

    abstract fun start()

    protected fun finish(result: Boolean) {
        this.result = result
        progressListener?.onStepFinish(this)
    }
}
